src/coreclr/jit/lclvars.cpp

```diff
@@ -2422,6 +2422,41 @@ void Compiler::lvaSetVarAddrExposed(unsigned varNum DEBUGARG(AddressExposedReaso
     lvaSetVarDoNotEnregister(varNum DEBUGARG(DoNotEnregisterReason::AddrExposed));
 }
 
+#ifdef DEBUG
+//------------------------------------------------------------------------
+// lvaSetHiddenBufferStructArg: Set the local var "varNum" as hidden buffer struct arg.
+//
+// Arguments:
+//    varNum - the varNum of the local
+//
+// TODO-ADDR-Bug: currently, we rely on these locals not being present in call argument lists,
+//   outside of the buffer address argument itself, as liveness - currently - treats the location node
+//   associated with the address itself as the definition point, and call arguments can be reordered
+//   rather arbitrarily. We should fix liveness to treat the call as the definition point instead and enable
+//   this optimization for "!lvIsTemp" locals.
+
```

Comment:
```suggestion
//------------------------------------------------------------------------
// lvaSetHiddenBufferStructArg: Set the local var "varNum" as hidden buffer struct arg.
//
// Arguments:
//    varNum - the varNum of the local
//
// Notes:
//    Most ABIs "return" large structures via return buffers, where the callee takes an address as the
//    argument, and writes the result to it. This presents a problem: ordinarily, addresses of locals
//    that escape to calls leave the local in question address-exposed. For this very special case of
//    a return buffer, however, it is known that the callee will not do anything with it except write
//    to it, once. As such, we handle addresses of locals that represent return buffers specially: we
//    *do not* mark the local address-exposed, instead saving the address' "Use*" in the call node, and
//    treat the call much like an "ASG(IND(addr), ...)" node throughout the compilation. A complicating
//    factor here is that the address can be moved to the late args list, and we have to fetch it from
//    the ASG setup node in that case. In the future, we should make it such that these addresses do
//    not ever need temps (currently they may because of conservative GLOB_REF setting on FIELD nodes).
//
//    TODO-ADDR-Bug: currently, we rely on these locals not being present in call argument lists,
//    outside of the buffer address argument itself, as liveness - currently - treats the location node
//    associated with the address itself as the definition point, and call arguments can be reordered
//    rather arbitrarily. We should fix liveness to treat the call as the definition point instead and
//    enable this optimization for "!lvIsTemp" locals.
//
```
Just adding some documentation.

Reply:
Thank you!

Reply:
Seems I force pushed and that erased these comments. I will resubmit it in next round.